test/polygon/raffle/HellsKitchen/HellsKitchen.test.ts

```diff
@@ -0,0 +1,440 @@
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+
+import {waitFor} from '../../../utils';
+
+import {
+  raffleSignWallet,
+  setupRaffle,
+  assert,
+  COLLECTION_MAX_SUPPLY,
+  contractName,
+} from './HellsKitchen.fixtures';
+
+// eslint-disable-next-line mocha/no-skipped-tests
+describe(contractName, function () {
+  it('should be able to mint with valid signature', async function () {
+    const {
+      raffleCollectionContract,
+      transferSand,
+      setupWave,
+      getNamedAccounts,
+      hre,
+      mint,
+    } = await setupRaffle();
+    const {deployer} = await getNamedAccounts();
+    await transferSand(deployer, '1000');
+    await setupWave(raffleCollectionContract, 20, 5, '10');
+    await mint(
+      raffleSignWallet,
+      deployer,
+      0,
+      raffleCollectionContract.address,
+      hre.network.config.chainId || 31337,
+      '10',
+      1
+    );
+
+    const transferEvents = await raffleCollectionContract.queryFilter(
+      raffleCollectionContract.filters.Transfer()
+    );
+
+    assert.equal(transferEvents.length, 1);
+  });
+
+  // eslint-disable-next-line mocha/no-skipped-tests
+  it(`should be able to mint ${COLLECTION_MAX_SUPPLY} different tokens`, async function () {
+    const {
+      raffleCollectionContract,
+      transferSand,
+      setupWave,
+      getNamedAccounts,
+      hre,
+      mint,
+    } = await setupRaffle();
+    const {deployer} = await getNamedAccounts();
+    await transferSand(deployer, '20000');
+    await setupWave(
+      raffleCollectionContract,
+      COLLECTION_MAX_SUPPLY,
+      COLLECTION_MAX_SUPPLY,
+      '1'
+    );
+    const tokens = [];
+    for (let i = 0; i < COLLECTION_MAX_SUPPLY; i++) {
```

Comment:
Is it necessary to mint all the tokens (with one tx / token) ? It's breaking the coverage tests because it's taking too much time

Reply:
maybe mint by batch ?

Reply:
There is a test to mint by batch and a test in one tx. Just to be sure, but we can skip it for now.

Reply:
implemented but only for Opensea OpenRegistry compliant and newer collections. The others will be skipped but that's in another pull request. Please accept this 
https://github.com/thesandboxgame/sandbox-smart-contracts/pull/894
then we can merge it here and have the final observations

Reply:
merge is done, you can rebase